Fail to build static library

System: CentOS 8

Fresh cloning the repository and running the following command `cargo test --features hdf5-sys/static,hdf5-sys/zlib` gave me this error:

```
Downloaded glob v0.3.1
  Downloaded serde_spanned v0.6.7
  Downloaded serde_derive v1.0.206
  Downloaded itoa v1.0.11
  Downloaded toml_datetime v0.6.8
  Downloaded termcolor v1.4.1
  Downloaded ryu v1.0.18
  Downloaded toml v0.8.19
  Downloaded trybuild v1.0.99
  Downloaded indexmap v2.3.0
  Downloaded serde v1.0.206
  Downloaded toml_edit v0.22.20
  Downloaded hashbrown v0.14.5
  Downloaded winnow v0.6.18
  Downloaded serde_json v1.0.124
  Downloaded 15 crates (986.7 KB) in 1.08s
   Compiling unicode-ident v1.0.12
   Compiling serde v1.0.206
   Compiling hashbrown v0.14.5
   Compiling equivalent v1.0.1
   Compiling serde_json v1.0.124
   Compiling winnow v0.6.18
   Compiling trybuild v1.0.99
   Compiling ryu v1.0.18
   Compiling itoa v1.0.11
   Compiling glob v0.3.1
   Compiling termcolor v1.4.1
   Compiling proc-macro2 v1.0.86
   Compiling proc-macro-error v1.0.4
   Compiling hdf5-metno-sys v0.9.2 (/data/kzhang/dev/hdf5-rust/hdf5-sys)
   Compiling quote v1.0.36
   Compiling syn v2.0.74
   Compiling indexmap v2.3.0
   Compiling proc-macro-error-attr v1.0.4
error: linking with `cc` failed: exit status: 1
  |
  = note: LC_ALL="C" PATH="/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin:/data/kzhang/micromamba/bin:/data/kzhang/micromamba/condabin:/data/kzhang/.cargo/bin:/data/kzhang/.local/bin:/data/kzhang/bin:/opt/slurm/sbin:/opt/slurm/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin" VSLANG="1033" "cc" "-m64" "/tmp/rustc8IvpPk/symbols.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.01lfbphb45qq7vjz1e4ckgpzx.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.0bt7cf8js3wg5ubyn3vwnvezd.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.1l69i9iaaloqmo0gm4fzedzb6.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.21wa2b91438yyjh7pk3g8tf6j.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.2a8qr7rrnxhc70rt9nb3ysl0p.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.3mp0fr9qlwi6mvik0a4vo9kfr.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.3rr9y3mk25rp5e1ibo8iyabqj.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.4c7rrmmnl4cs4kc8pl3om9ke6.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.4routw1dr8a2uqtf71fy75k1p.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.4xblcvgu31k12te0z2yag89mt.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.54cpp0q35q1jl9hszski3fvfm.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.5w5bmgd4bov4a1sxf285mydiq.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.5xptiplmvyq7eh9kue0q99rh7.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.6xkjsfed982fk8rbps7ovldz1.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.6zf7gyzgubalsdsg820m7pd5f.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.7p5mytne8bitgilm6si35gt3g.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.7rcdtlf554a1ycj6k6xif3j0f.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.85jst6xy6azvk99gdy5nrgwta.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.8frp1vq5tfswa761306q7raa1.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.8pd4kfox1w6b97si6iphg9h73.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.8ylg97fup9y06ghnfjtpgutek.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.9lqchiq2baplw2hjpl06slhnn.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.a9yeff35qp61x1ji501we6wir.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.b7lwhqvil4e8wujqsfy4wbv6l.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.bi86laymkwfddq1h3589gw6l2.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.cpsci8zltdd3oy8ae607dzomi.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.cvnfrpynkcpb54482qq0ni8h1.rcgu.o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8.dy5il5zoxq087d14738gzax5y.rcgu.o" "-Wl,--as-needed" "-L" "/data/kzhang/dev/hdf5-rust/target/debug/deps" "-L" "/data/kzhang/dev/hdf5-rust/target/debug/build/hdf5-metno-src-40f7dcbef59f7ad7/out/lib" "-L" "/data/kzhang/dev/hdf5-rust/target/debug/build/libz-sys-6a5ed2bc42eb92ca/out/lib" "-L" "/data/kzhang/dev/hdf5-rust/target/debug/build/libz-sys-6a5ed2bc42eb92ca/out/lib" "-L" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "-Wl,--whole-archive" "-lz" "-Wl,--no-whole-archive" "-Wl,--whole-archive" "-lhdf5_debug" "-Wl,--no-whole-archive" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libtest-95407a28b34cb7d8.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgetopts-6cf033bd0623f965.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunicode_width-92be448731e25aff.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_std-9c68449f4da60201.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/libparking_lot-c411f61bdce3d6f1.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/libparking_lot_core-84ea998473dda687.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/libcfg_if-5d8cf5fd7778c4e1.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/libsmallvec-e253e2b315ed3fe5.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/liblock_api-1db850655e0663b3.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/libscopeguard-e9d23efdb2d1a97e.rlib" "/data/kzhang/dev/hdf5-rust/target/debug/deps/liblibc-36f95228859c41d7.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-52417a9a08ba8fb9.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-1b238e97929fdc18.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-3692c2663daa48a5.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libmemchr-5c1e2d61611c0240.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-e55c5186b9bbabcb.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-7911d2fe92903c6c.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-c03510cdc48cfa52.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd_detect-7ec6f3765287c7b7.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-ab3f1b788cf79cfc.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-6db4226b4e34b1a6.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-f5bb1a38d410fa1e.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-e585efe74c64604d.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-03f8e93b59ffc316.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-669e9182bfe197a6.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-64b53226be979181.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-26921ac7e6a44d2b.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-6703049be165ebf1.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-04bfdf6b094564ce.rlib" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-6648dc218e522d87.rlib" "-Wl,-Bdynamic" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-L" "/data/kzhang/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-o" "/data/kzhang/dev/hdf5-rust/target/debug/deps/hdf5_metno_sys-6ca0177c2018caf8" "-Wl,--gc-sections" "-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs"
  = note: /data/kzhang/dev/hdf5-rust/target/debug/build/hdf5-metno-src-40f7dcbef59f7ad7/out/lib/libhdf5_debug.a(H5Z.c.o): In function `H5Z_init':
          /data/kzhang/dev/hdf5-rust/hdf5-src/ext/hdf5/src/H5Z.c:113: undefined reference to `SZ_encoder_enabled'
          /data/kzhang/dev/hdf5-rust/target/debug/build/hdf5-metno-src-40f7dcbef59f7ad7/out/lib/libhdf5_debug.a(H5Zszip.c.o): In function `H5Z__filter_szip':
          /data/kzhang/dev/hdf5-rust/hdf5-src/ext/hdf5/src/H5Zszip.c:298: undefined reference to `SZ_BufftoBuffDecompress'
          /data/kzhang/dev/hdf5-rust/hdf5-src/ext/hdf5/src/H5Zszip.c:325: undefined reference to `SZ_BufftoBuffCompress'
          collect2: error: ld returned 1 exit status

  = note: some `extern` functions couldn't be found; some native libraries may need to be installed or have their path specified
  = note: use the `-l` flag to specify native libraries to link
  = note: use the `cargo:rustc-link-lib` directive to specify the native libraries to link with Cargo (see https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-link-lib)

error: could not compile `hdf5-metno-sys` (lib test) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

--- a/hdf5-src/Cargo.toml
+++ b/hdf5-src/Cargo.toml
@@ -22,7 +22,7 @@ exclude = [
     "ext/hdf5/HDF5Examples/**",
     "ext/hdf5/doxygen/**",
 ]
-version.workspace = true
+version = "0.9.1"  # !V
 rust-version.workspace = true
 authors.workspace = true
 keywords.workspace = true

--- a/hdf5-src/build.rs
+++ b/hdf5-src/build.rs
@@ -32,6 +32,7 @@ fn main() {
         "ALLOW_UNSUPPORTED",
         "HDF5_BUILD_HL_LIB",
         "HDF5_ENABLE_NONSTANDARD_FEATURE_FLOAT16",
+        "HDF5_ENABLE_SZIP_SUPPORT",
     ] {
         cfg.define(option, "OFF");
     }

